src/common/meta/src/wal/kafka/topic_selector.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use serde::{Deserialize, Serialize};
+
+use crate::wal::kafka::topic::Topic;
+
+/// The type of the topic selector, i.e. with which strategy to select a topic.
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
```

Comment:
I think the name should be consistent with selector type in metasrv

https://docs.greptime.com/developer-guide/metasrv/selector#selector-type

Reply:
Topic manager now initializes the topic selector according to the given config.

Reply:
The selector name is changed to `xxxBased`. For e.g. `RoundRobinBased`.